xbmc/addons/AddonInfo.h

```diff
@@ -85,13 +85,240 @@ namespace ADDON
   typedef std::map<std::string, std::string> InfoMap;
   typedef std::map<std::string, std::string> ArtMap;
 
+  class CAddonBuilder;
+
   class CAddonInfo
   {
   public:
     CAddonInfo();
     CAddonInfo(std::string id, TYPE type);
 
     /*!
+     * @brief To ask generated class is usable and all needed parts are set.
+     *
+     * @return True is OK and usable, otherwise false.
+     *
+     * @note This function is only be related to constructor's who read the
+     * addon xml part.
+     */
+    bool IsUsable() const { return m_usable; }
+
+    /*!
+     * @brief Parts used from ADDON::CAddonDatabase
+     */
+    //@{
+    void SetInstallDate(CDateTime installDate) { m_installDate = installDate; }
+    void SetLastUpdated(CDateTime lastUpdated) { m_lastUpdated = lastUpdated; }
+    void SetLastUsed(CDateTime lastUsed) { m_lastUsed = lastUsed; }
+    void SetOrigin(std::string origin) { m_origin = std::move(origin); }
+    //@}
```

Comment:
Also a part who becomes in future used from CAddonDatabase. Can remove it and add later.